src/main/java/com/app/onlinebookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+import com.app.onlinebookstore.dto.UpdateCartItemDto;
+import com.app.onlinebookstore.exaption.EntityNotFoundException;
+import com.app.onlinebookstore.mapper.CartItemMapper;
+import com.app.onlinebookstore.model.CartItem;
+import com.app.onlinebookstore.model.ShoppingCart;
+import com.app.onlinebookstore.model.User;
+import com.app.onlinebookstore.repository.BookRepository;
+import com.app.onlinebookstore.repository.CartItemRepository;
+import com.app.onlinebookstore.repository.ShoppingCartRepository;
+import com.app.onlinebookstore.repository.UserRepository;
+import com.app.onlinebookstore.service.CartItemService;
+import com.app.onlinebookstore.service.UserService;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final BookRepository bookRepository;
+    private final UserService userService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    @Transactional
+    public CartItemDto save(CreateCartItemRequestDto requestDto) {
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(bookRepository.findById(requestDto.getBookId()).get());
+        cartItem.setQuantity(requestDto.getQuantity());
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Optional<User> user = userRepository.findByEmail(authentication.getName());
+        Long userId = user.get().getId();
+        setShoppingCartAndCartItems(shoppingCartRepository
+                .findByUserId(userId).get().getId(), cartItem);
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public CartItemDto update(Long id, UpdateCartItemDto updateCartItemDto) {
+        CartItem cartItem = cartItemRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find item by id " + id));
+        cartItem.setQuantity(updateCartItemDto.getQuantity());
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public List<CartItemDto> findCartItemsByShoppingCarts(Long id) {
+        return cartItemRepository.findCartItemsByShoppingCartId(id).stream()
+                .map(cartItemMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void delete(Long id) {
+        cartItemRepository.deleteById(id);

```

Comment:
What if user wants to delete item which doesn't belong to him? Make sure you are securing your APIs not only by endpoint access control, but also by resource access. Make sure to add this type of security to all serivces where it is needed